data_diff/database.py

```diff
@@ -261,11 +380,45 @@ def quote(self, s: str):
     def to_string(self, s: str):
         return f"cast({s} as varchar(1024))"
 
+    def select_table_schema(self, path: DbPath) -> str:
+        if len(path) > 1:
+            raise ValueError("Unexpected table path for oracle")
+        (table,) = path
+
+        return (
+            f"SELECT column_name, data_type, 9 as datetime_precision, data_precision as numeric_precision"
+            f" FROM USER_TAB_COLUMNS WHERE table_name = '{table.upper()}'"
+        )
+
+    def normalize_value_by_type(self, value, coltype: ColType) -> str:
+        if isinstance(coltype, PrecisionType):
+            return f"to_char(cast({value} as timestamp), 'YYYY-MM-DD HH24:MI:SS.FF{coltype.precision or ''}')"
```

Comment:
Why do you need to convert to `to_char` here? Doesn't the wrapping `concat` do that automatically?

Reply:
`to_char()` does the date formatting.

Of course it does. It's called to_char(), what else could it do?

Reply:
LOL